File: Recode3/ListAndNotification/app/src/main/java/com/example/listandnotification/service/CursoService.java
package com.example.listandnotification.service;

import com.example.listandnotification.model.Curso;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;

public interface CursoService {

    @GET("course")
    Call<List<Curso>> getAllCourses();

    @POST("course")
    Call<Curso> insert(@Body Curso curso);



    @GET("course")
    Call<Curso> getCurso();

    @PUT("course")
    void update();

    @DELETE("course")
    void delete();

}

File: ProfessorAllocation/app/src/main/java/com/domain/professorallocation/CursoService.kt
package com.domain.professorallocation

import retrofit2.Call
import retrofit2.http.*

interface CursoService {

    @GET("courses")
    fun getAll(): Call<List<Curso>>

    @POST("courses")
    fun cadCurso(@Body curso: Curso): Call<Curso>

    @PUT("courses/{id}")
    fun alterCurso(@Path("id") cursoId: String, @Body curso: CursoPutDTO): Call<Curso>

    @DELETE("courses/{id}")
    fun deleteCurso(@Path("id") cursoId: Int)

}